src/context/AuthContext.jsx

```diff
@@ -0,0 +1,108 @@
+import {createContext, useEffect, useState} from "react";
+import axios from "axios";
+import {jwtDecode} from "jwt-decode";
+import isTokenValid from "../helpers/isTokenValid";
+
+export const AuthContext = createContext(null);
+
+
+function AuthContextProvider({children}) {
+    const [auth, setAuth] = useState({
+        isAuth: false,
+        user: null,
+        status: 'pending',
+        token: '',
+    });
+
+    // PERSIST ON REFRESH //
+    useEffect(() => {
+        // is er een token?
+        // zo ja, probeer te decoderen en check geldigheid.
+        //// gelukt en geldig? login()
+        //// lukt dat niet en/of niet geldig: logout()
+
+        // geen token: logout()
+        const validateToken = async () => {
+
+            const token = localStorage.getItem('token');
+            if (token) { // er is een token gevonden
+                try {
+                    const decodedToken = jwtDecode(token);
+
+                    if (isTokenValid(decodedToken)) {
+                        await login(token);  // login als token valid is
```

Comment:
Dit hoeft geen functie te zijn omdat er geen asynchrone processen in plaatsvinden. Dus je kunt de if-statement direct in de useEffect zetten! Ook kun je je leven een stukje makkelijker maken door de isTokenValid-helper verantwoordelijk te maken voor het decoderen van de token, zodat je het korter op kunt schrijven:

`if (token && isTokenValid(token)) { void login(token); } else { logout(); }`